prebuilt/core/booking.json

```diff
@@ -827,7 +827,49 @@
         "MODE_RAIL": {
           "$schema": "http://json-schema.org/draft-04/schema#",
           "description": "Schema for MODE_RAIL meta field",
-          "type": "object"
+          "type": "object",
+          "properties": {
+            "outward": {
+              "stationId": {
+                "type": "string",
+                "minLength": 1,
+                "maxLength": 64
+              },
+              "stationName": {
```

Comment:
I'd split this into two: stationFullName (stationLongName) and stationShortName.

In NS there are 3 identifiers used in different context: id, long name, and abbreviation. I guess abbreviation could go to short name or actually there should be abbreviation name.

Reply:
Moreover, if we design this from scratch European stations use UIC which is basically ID.

Here is an example data from NS:
```
{
            "uicCd": 118400426,
            "cd": 426,
            "naamKort": "Maastr. Randwyck",
            "naam": "Maastricht Randwyck",
            "verkorting": "mtr",
            "finCd": 84426,
            "cdLand": "NL",
            "vervoerder": [
                {
                    "cdVervoerder": 64
                },
                {
                    "cdVervoerder": 2
                }
            ],
            "geoInfo": {
                "latitude": 50.8386116027832,
                "longitude": 5.71722221374512,
                "radius": 200,
                "naderenRadius": 1200
            }
        },
```

Here you can see UIC code, `cd` - is an internal NS id (which in most cases is the last 3 digits of UIC). You have 3 names - long, short and abbreviation (`verkorting`). In there response you can also see `cdVervoerder` which tells you what carriers (operators) use this station (in this case 64 = NS, 2 = Arriva).

Reply:
Additionally, you could add route details. In NS this is basically:

- fromStation
- toStation
- carrierCode
- carrierName,
- cost

Reply:
> In NS there are 3 identifiers used in different context: id, long name, and abbreviation.

Can't these be translated to only name + id ?
Which one is used to fetch station meta data, id or abbreviation? That will be stationId

Reply:
> Cost is a first class property for booking

Yes, I was thinking about the individual cost of each leg. So the first class property is total cost.

Reply:
> Can't these be translated to only name + id ?

That's complicated ;) I agree with id and name. NS, additionally, uses abbreviation while generating tickets.

Reply:
name for NS station can then be `Abbreviation - Long name`
you can split that in TSP adapter

Reply:
@blackevil245 Are you saying to keep abbreviation and long name in one field like: "mtr - Maastricht Randwyck"? Looks to hacky for me and then you need to split it or use some regex.

Reply:
I think it is ok as a hack. On the good side, clients are using leg.name already, so no need for extra works to display abbreviations on clients

Reply:
I don't need to display abbreviation on clients. It is used to generate tickets. I'd prefer to have additional property instead of nesting everything in one. Otherwise, you could scarp the whole JSON and use CSV and say just split the values :)

Reply:
@blackevil245 BTW What about keeping partial cost (cost of individual legs). The top-level property is total cost. Where would you store leg cost?

Reply:
> I don't need to display abbreviation on clients. It is used to generate tickets.

If the info is bound to logic rather than display, then why don't we use id / stationId for that? NS use abbreviation instead of id for their operation, does that mean their id is meaningless? If so, you can use abbreviation as the ID. No point in storing data from TSP that we don't have any use of.

> What about keeping partial cost (cost of individual legs)

I thought it's discussed that this is what we need only in the daily report? If so we don't need to store it in booking schema

Reply:
> we don't need to store it in booking schema

Oh. yes. My mistake. I mixed it up.

> does that mean their id is meaningless? 

You need to go and check with NS :) Their API is rather weird... Anyway, I was thinking that once I have that abbreviation I can keep it and re-use it. There are other ways to achieve the same result. I guess I need to re-think/re-design that bit.  Nevertheless, it's a topic for another discussion.

Reply:
> You need to go and check with NS :)

Yeah definitely. I can come join when you have workshop / meeting f2f with them. Since they don't really have interfaces ready, i think we can suggest changes :)